### DISCUSSION TITLE
Best practice for embedding known keywords?

### DISCUSSION THREAD

**Discussion by zilch42** [MaartenGr/BERTopic#1044] [Q&A] [answered]
Hi there,

Just wondering if there is a good way (or if it is even good practice) to include known keywords with the text to embed in the sentence embedding model? E.g. say I had journal articles each of which had a Title, Abstract and a list of Keywords supplied by the author. Those Keywords contain a lot of useful information about the topic but aren't in a sentence structure. Would it be good to include each keyword as a sentence (separated by ". ") to the embedding model?

e.g. Doc = [Title]. [Keyword 1]. [Keyword 2]. [Keyword 3]... [Keyword n]. [Abstract]

Will the order affect the interpretation or is each sentence in the document considered independent?

**Comment by MaartenGr** [comment, marked as answer]
There is no one right method to do this or any good practice as it depends on a number of things. First, the embedding model that you choose will be important here. If you are combining titles, keywords, and abstracts, then the resulting document might become too long and not fit within the embedding models token limit. There are others that can handle this better, like longformer, but there is a trick that might be worthwhile here. 

You can embed the title, keywords, and abstract independently from one another and average them together. Here, you could even decide to do a weighted average as the abstract might be more informative than the keywords.

**Reply by zilch42** [reply]
Thanks that sounds like a great tip! Some of my abstracts are already too long and I hadn't yet come to thinking about how to handle that. A weighting schema is a great idea too!
